Python 3.11 - Building loadgen has issues with pinned third_party/pybind version

When I follow the loadgen build instructions, I get:

```
../third_party/pybind/include/pybind11/cast.h:442:36: error: invalid use of incomplete type ‘PyFrameObject’ {aka ‘struct _frame’}
        442 |                 "  " + handle(frame->f_code->co_filename).cast<std::string>() +
            |                                    ^~
      In file included from /opt/conda/envs/firesim/include/python3.11/Python.h:42,
                       from ../third_party/pybind/include/pybind11/detail/common.h:112,
                       from ../third_party/pybind/include/pybind11/pytypes.h:12,
                       from ../third_party/pybind/include/pybind11/cast.h:13,
                       from ../third_party/pybind/include/pybind11/attr.h:13,
                       from ../third_party/pybind/include/pybind11/pybind11.h:44,
                       from ../third_party/pybind/include/pybind11/functional.h:12,
                       from bindings/python_api.cc:27:
```

I looked up this issue online, and found that this was a change in Python 3.11: [https://github.com/pybind/pybind11/discussions/4333](https://github.com/pybind/pybind11/discussions/4333)

It has already been fixed in pybind11; however, the mlcommons/inference project has an old (~5 years) pybind version [pinned](https://github.com/mlcommons/inference/tree/master/third_party) in [third_party/pybind](https://github.com/pybind/pybind11/tree/25abf7efba0b2990f5a6dfb0a31bc65c0f2f4d17)

By pulling the latest pybind version, I was able to compile loadgen with no problems.

Is there a reason for this (e.g. consistency with old versions)? If not, I would suggest either:
- Update the pinned pybind version to a more recent release, so that Python 3.11 ppl can compile loadgen easier
- Put in the install/build instructions a notice that Python 3.11 will require updating pybind

Thanks!

--- a/loadgen/setup.py
+++ b/loadgen/setup.py
@@ -77,7 +77,7 @@ mlperf_long_description = (this_directory / "README.md").read_text(encoding="utf
 mlperf_loadgen_module = Pybind11Extension(
         "mlperf_loadgen",
         define_macros=[("MAJOR_VERSION", "3"), ("MINOR_VERSION", "1")],
-        include_dirs=[".", "../third_party/pybind/include"],
+        include_dirs=[".", get_include()],
         sources=mlperf_loadgen_sources,
         depends=mlperf_loadgen_headers)
 

--- a/loadgen/setup.py
+++ b/loadgen/setup.py
@@ -28,6 +28,8 @@ from setuptools import Extension
 from setuptools import setup
 from version_generator import generate_loadgen_version_definitions
 from pathlib import Path
+from pybind11 import get_include
+from pybind11.setup_helpers import Pybind11Extension, build_ext
 
 generated_version_source_filename = "generated/version_generated.cc"
 generate_loadgen_version_definitions(generated_version_source_filename, ".")
@@ -72,7 +74,7 @@ mlperf_loadgen_sources = (mlperf_loadgen_sources_no_gen +
 mlperf_long_description = (this_directory / "README.md").read_text(encoding="utf-8")
 
 
-mlperf_loadgen_module = Extension(
+mlperf_loadgen_module = Pybind11Extension(
         "mlperf_loadgen",
         define_macros=[("MAJOR_VERSION", "3"), ("MINOR_VERSION", "1")],
         include_dirs=[".", "../third_party/pybind/include"],
@@ -83,6 +85,7 @@ setup(name="mlperf_loadgen",
       version="3.1",
       description="MLPerf Inference LoadGen python bindings",
       url="https://mlcommons.org/",
+      cmdclass={"build_ext": build_ext},
       ext_modules=[mlperf_loadgen_module],
       long_description=mlperf_long_description,
       long_description_content_type='text/markdown')

